sdk/python/feast/offline_store.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List, Optional, Tuple
+
+import pyarrow
+
+from feast.feature_view import FeatureView
+
+
+class OfflineStore(ABC):
+    """
+    OfflineStore is a non-user-facing object used for all interaction between Feast and the service used for offline storage of features. Currently BigQuery is supported.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def pull_latest_from_table(
+        table_ref: str,
+        entity_names: List[str],
+        feature_names: List[str],
+        event_timestamp_column: str,
+        created_timestamp_column: Optional[str],
+        start_date: datetime,
+        end_date: datetime,
+    ) -> Optional[pyarrow.Table]:
+        pass
+
+
+class BigQueryOfflineStore(OfflineStore):
+    """
+    BigQueryOfflineStore is a non-user-facing object used for all interaction between Feast and BigQuery.
```

Comment:
I think we can maybe just leave out this comment and inherit from OfflineStore for now.